src/controllers/payByAccountDetails.controller.ts

```diff
@@ -25,18 +31,28 @@ export class PayByAccountDetailsController extends BaseController {
   private readonly ERROR_INCORRECT_CREDENTIALS: string = 'Your Presenter ID or Presenter authentication code is incorrect'
 
   public constructor(
-    @inject(TYPES.PAY_BY_ACCOUNT_FEATURE_ENABLED) private PAY_BY_ACCOUNT_FEATURE_ENABLED: number,
+    @inject(SessionService) private sessionService: SessionService,
+    @inject(DissolutionService) private dissolutionService: DissolutionService,
     @inject(FormValidator) private validator: FormValidator,
-    @inject(PayByAccountService) private payByAccountService: PayByAccountService) {
+    @inject(PayByAccountService) private payByAccountService: PayByAccountService,
+    @inject(TYPES.PAY_BY_ACCOUNT_FEATURE_ENABLED) private PAY_BY_ACCOUNT_FEATURE_ENABLED: number
+  ) {
     super()
   }
 
   @httpGet('')
-  public async get(): Promise<string> {
+  public async get(): Promise<string | RedirectResult> {
     if (!this.PAY_BY_ACCOUNT_FEATURE_ENABLED) {
       return Promise.reject(new NotFoundError('Feature toggle not enabled'))
     }
 
+    const token: string = this.sessionService.getAccessToken(this.httpContext.request)
+    const dissolutionSession: DissolutionSession = this.sessionService.getDissolutionSession(this.httpContext.request)!
+
+    if (await this.isAlreadyPaid(dissolutionSession, token)) {
+      return this.redirect(SEARCH_COMPANY_URI)
+    }
+
```

Comment:
could the token about session fetching go into the `isAlreadyPaid` method as that's the only place they're used